UVOL 2.0

the-reality-engineer opened the pull request.

the-reality-engineer committed:
uvol-2-alpha

the-reality-engineer committed:
Add blender scripts

CITIZENDOT committed:
Added demo sub-project and new manifest format

CITIZENDOT committed:
Minor fix: decoder.wasm wasn't being added to dist

CITIZENDOT committed:
Loading manifest before video

CITIZENDOT committed:
Encoder improvements

CITIZENDOT committed:
Added sample files

CITIZENDOT committed:
Added lossless draco sample files

CITIZENDOT committed:
Backward-Compatible KTX2 player

CITIZENDOT committed:
Fixed KTX2 memory issues and added 1k texture demo

CITIZENDOT committed:
Added progress bars to encoder

CITIZENDOT committed:
Added encoder docs

CITIZENDOT committed:
Updated fetchBuffers

CITIZENDOT committed:
Fixed buffering issues and added support for videos with no audio

CITIZENDOT committed:
Fix new track overflow bug

CITIZENDOT committed:
Support for varying frame rates

CITIZENDOT committed:
Use only one ShaderMaterial all the time

the-reality-engineer committed:
Update README.md

the-reality-engineer committed:
Update README.md

CITIZENDOT committed:
create template config & warn about non-divisible frame rates

the-reality-engineer commented on src/V2/player.ts:
```diff
@@ -0,0 +1,461 @@
+        if (this.fileHeader.AudioURL) {
+            this.currentTime = this.audio.currentTime;
+        } else {
+            this.currentTime += this.clock.getDelta()
```
@CITIZENDOT  We shouldn't use getDelta() here, we should just use Date.now() - playbackStartTime

CITIZENDOT replied:
I was using [THREE.Clock](https://threejs.org/docs/#api/en/core/Clock) here, that uses perfomance.now internally when available. Shall I replace time handling with Date?

CITIZENDOT commented:
@cla-bot check

CITIZENDOT committed:
Added usage section and fixed encoder bug

CITIZENDOT committed:
update start function and manifest in demo

CITIZENDOT committed:
Fix pattern bug and typescript warning

CITIZENDOT committed:
Backward compatibility at track level

kungfooman commented:
Given the manifest:

```
{
    "Version": "v2",
    "DRCURLPattern": "LIAM_DRC/export_#####.obj.drc",
    "KTX2URLPattern": "LIAM_KTX2/export_#####.1k.ktx2",
    "AudioURL": "liamlow.mp3",
    "BatchSize": 5,
    "GeometryFrameCount": 250,
    "TextureSegmentCount": 50,
    "TotalFrames": 250,
    "GeometryFrameRate": 25,
    "TextureFrameRate": 25
}
```

Can we extract a bounding box from `drc` or `ktx2` without recalculating it by min/max-ing every vertex?

Otherwise we still don't have a AABB / bounding box for meshes, which is IMO the biggest performance/usability issue.

kungfooman commented on src/V2/player.ts:
```diff
@@ -79,6 +79,7 @@ export default class Player {
     private vertexShader: string
     private fragmentShader: string
+    //@ts-ignore
     private intervalId: NodeJS.Timer
```
Isn't that just a number? Why overcomplicate it twice?

CITIZENDOT replied:
Yes. I'll make it a number.

CITIZENDOT commented:
> 

I'm not sure, but I don't think that's possible, since they're in compressed formats. Btw what are the advantages by precalculating it? When extracting the draco and ktx2 frames, `boundingSphere` is still calculated anyway, na? (please correct me If I'm wrong)

kungfooman commented:
> I'm not sure if that's possible, since they're in compressed formats. Btw what are the advantages by precalculating it? When extracting the draco and ktx2 frames, `boundingSphere` is still calculated anyway, na? (please correct me If I'm wrong)

It's important to have aabb info as soon as possible, so you can properly focus an OrbitCamera e.g. onto it, while it's still loading (or not loaded at all, just the manifest - lets say in an editor where you layout your scene graph).

Other volumetric video formats give you the AABB straight away for that reason.

CITIZENDOT committed:
Added consistent pause across both versions & use Date for handling time

CITIZENDOT committed:
Seperate V1/V2 args and fix encoder start file issue

CITIZENDOT committed:
Shader fixes

- Fixed shaders to work with logarthmicDepthBuffer
- Set manifest path at `setTrackPath` instead of accessing from `paths`
- Check `onMeshBuffering` existence before calling
- Use `video` element for `audio` source

CITIZENDOT committed:
typo

CITIZENDOT committed:
added paused and stopped utilities

CITIZENDOT committed:
minor pause play bugs

CITIZENDOT committed:
fixed v2 pause/play, set fileHeader to null after track end

CITIZENDOT committed:
Add stats to demo player

CITIZENDOT committed:
Refactored to work with new manifest format

CITIZENDOT committed:
Added manifest

CITIZENDOT committed:
remove corto binary

CITIZENDOT committed:
Merge branch 'dev' into uvol-2-alpha

the-reality-engineer committed:
release 2.0.0

CITIZENDOT commented:
> Given the manifest:
> 
> ```
> {
>     "Version": "v2",
>     "DRCURLPattern": "LIAM_DRC/export_#####.obj.drc",
>     "KTX2URLPattern": "LIAM_KTX2/export_#####.1k.ktx2",
>     "AudioURL": "liamlow.mp3",
>     "BatchSize": 5,
>     "GeometryFrameCount": 250,
>     "TextureSegmentCount": 50,
>     "TotalFrames": 250,
>     "GeometryFrameRate": 25,
>     "TextureFrameRate": 25
> }
> ```
> 
> Can we extract a bounding box from `drc` or `ktx2` without recalculating it by min/max-ing every vertex?
> 
> Otherwise we still don't have a AABB / bounding box for meshes, which is IMO the biggest performance/usability issue.

Thanks for this suggestion! It's work in progress now.
